### PR TITLE
HSEARCH-2715 Mention in the documentation *when* a full reindex may be necessary

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-2715

### CODE REVIEW

File: documentation/src/main/asciidoc/manual-index.asciidoc [hunk 1]
```diff
@@ -100,8 +100,8 @@ successfully committed, or you make use of flushToIndexes.
 [[search-batchindex]]
 === Rebuilding the whole index
 
-If you change the entity mapping to the index, chances are that the whole Index needs to be updated;
-For example if you decide to index a an existing field using a different analyzer you'll need to
+If you change the entity mapping to the index, chances are that the whole index needs to be updated;
+for example if you decide to index a an existing field using a different analyzer you'll need to
```
Comment (gsmet) [inline documentation/src/main/asciidoc/manual-index.asciidoc]: s/a an/an/

File: documentation/src/main/asciidoc/manual-index.asciidoc [hunk 2]
```diff
@@ -111,6 +111,20 @@ Hibernate Search provides two main strategies to choose from:
...
 
+[NOTE]
+====
+When you change the entity mapping to the index, if you don't know whether the changes require
+a full reindexing or not, you can follow this rule of thumb:
+
+* Rebuilding the whole index is *not* required if all you did was:
+** *adding* new indexes/fields that will be empty for all currently persisted entities,
+   e.g. adding a new property on an entity type and mapping it to a field,
+   but with the guarantee that this property will initially be null for every instance of this entity;
```
Comment (gsmet) [inline documentation/src/main/asciidoc/manual-index.asciidoc]: I think it would be nice to explicitly add the case when you add new entities.

**Comment by yrodiere** [comment]
@gsmet Right. I amended my commit.

**Comment by gsmet** [comment]
Rebased and merged. Thanks.

**Pull request closed without merging**